src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,175 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/replace_symbolic.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+
+
+namespace LCompilers {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+
+class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisitor>
+{
+public:
+    ReplaceSymbolicVisitor(Allocator &al_) :
+    PassVisitor(al_, nullptr) {
+        pass_result.reserve(al, 1);
+    }
+
+    bool symbolic_replaces_with_CPtr_Module = false;
+    bool symbolic_replaces_with_CPtr_Function = false;
+
+    void visit_Module(const ASR::Module_t &x) {
+        SymbolTable* current_scope_copy = current_scope;
+        current_scope = x.m_symtab;
+        for (auto &a : x.m_symtab->get_scope()) {
+            this->visit_symbol(*a.second);
+            if(symbolic_replaces_with_CPtr_Module){
+                std::string new_name = current_scope->get_unique_name("basic_new_stack");
+                if(current_scope->get_symbol(new_name)) return;
+                std::string header = "symengine/cwrapper.h";
+                SymbolTable *fn_symtab = al.make_new<SymbolTable>(current_scope);
+
+                Vec<ASR::expr_t*> args;
+                {
+                    args.reserve(al, 1);
+                    ASR::ttype_t *arg_type = ASRUtils::TYPE(ASR::make_CPtr_t(al, x.base.base.loc));
+                    ASR::symbol_t *arg = ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                        al, x.base.base.loc, fn_symtab, s2c(al, "x"), nullptr, 0, ASR::intentType::In,
+                        nullptr, nullptr, ASR::storage_typeType::Default, arg_type, nullptr,
+                        ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+                    fn_symtab->add_symbol(s2c(al, "x"), arg);
+                    args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, arg)));
+                }
+                Vec<ASR::stmt_t*> body;
+                body.reserve(al, 1);
+
+                Vec<char *> dep;
+                dep.reserve(al, 1);
+
+                ASR::asr_t* new_subrout = ASRUtils::make_Function_t_util(al, x.base.base.loc,
+                    fn_symtab, s2c(al, new_name), dep.p, dep.n, args.p, args.n, body.p, body.n,
+                    nullptr, ASR::abiType::BindC, ASR::accessType::Public,
+                    ASR::deftypeType::Interface, nullptr, false, false, false,
+                    false, false, nullptr, 0, nullptr, 0, false, false, false, s2c(al, header));
```

Comment:
```suggestion
                    false, false, nullptr, 0, nullptr, 0, false, false, false);
```
Not required, I think. Does it fail without it?

Reply:
Well locally everything seems to build perfectly , so I am not sure why build is failing here . 

I guess we would need that right ?
```
                            basic_new_stack:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            x:
                                                (Variable
                                                    6
                                                    x
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (CPtr)
                                                    ()
                                                    BindC
                                                    Public
                                                    Required
                                                    .true.
                                                )
                                        })
                                    basic_new_stack
                                    (FunctionType
                                        [(CPtr)]
                                        ()
                                        BindC
                                        Interface
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 6 x)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    "symengine/cwrapper.h"
                                )
```
So that we could pass the headerfile `symengine/cwrapper.h` ?
I see that it is assigned as `nullptr` by default 
```
inline ASR::asr_t* make_Function_t_util(Allocator& al, const Location& loc,
    SymbolTable* m_symtab, char* m_name, char** m_dependencies, size_t n_dependencies,
    ASR::expr_t** a_args, size_t n_args, ASR::stmt_t** m_body, size_t n_body,
    ASR::expr_t* m_return_var, ASR::abiType m_abi, ASR::accessType m_access,
    ASR::deftypeType m_deftype, char* m_bindc_name, bool m_elemental, bool m_pure,
    bool m_module, bool m_inline, bool m_static, ASR::ttype_t** m_type_params, size_t n_type_params,
    ASR::symbol_t** m_restrictions, size_t n_restrictions, bool m_is_restriction,
    bool m_deterministic, bool m_side_effect_free, char *m_c_header=nullptr) {
    ASR::ttype_t* func_type = ASRUtils::TYPE(ASRUtils::make_FunctionType_t_util(
        al, loc, a_args, n_args, m_return_var, m_abi, m_deftype, m_bindc_name,
        m_elemental, m_pure, m_module, m_inline, m_static, m_type_params, n_type_params,
        m_restrictions, n_restrictions, m_is_restriction));
    return ASR::make_Function_t(
        al, loc, m_symtab, m_name, func_type, m_dependencies, n_dependencies,
        a_args, n_args, m_body, n_body, m_return_var, m_access, m_deterministic,
        m_side_effect_free, m_c_header);
}
```

Reply:
I think, we require this for C backend, Let's keep it!